src/track/beatmap.cpp

```diff
@@ -302,9 +317,9 @@ double BeatMap::findNthBeat(double dSamples, int n) const {
         return -1;
     }
 
-    Beat beat;
-    // Reduce sample offset to a frame offset.
-    beat.set_frame_position(static_cast<google::protobuf::int32>(samplesToFrames(dSamples)));
+    Beat beat = beatFromFramePos(
+            mixxx::audio::FramePos::fromEngineSamplePos(dSamples)
+                    .toFullFrames());
```

Comment:
This changes the code for negative values because of internal floor()

Reply:
If negative values are not allowed we should check for that. Otherwise rounding to frame boundaries should always be done in the same direction. Depending on the use case but independent of the sign. Rounding in different directions is probably wrong.

Reply:
I was wondering if `std::trunc` is preferable because it always rounds towards zero. But no idea.

Reply:
On second thought: trunc may lead to tiny Bpm changes around zero because the distance between beats is smaller. So using floor is the right call IMHO.

Reply:
Yes right.

Reply:
Strictly speaking we have now a new beat map version. 
But since we have now beats below zero it does not matter.